forwarder/pkg/common/egress_interface.go

```diff
@@ -78,25 +78,36 @@ func parseProcFile(scanner *bufio.Scanner) (string, net.IP, error) {
 		if strings.TrimSpace(parts[1]) == "00000000" {
 			outgoingInterface := strings.TrimSpace(parts[0])
 			defaultGateway := strings.TrimSpace(parts[2])
-			ip := parseGatewayIP(defaultGateway)
+			ip, err := parseGatewayIP(defaultGateway)
+			if err != nil {
+				logrus.Errorf("error processing gateway IP %v for outgoing interface: %v", defaultGateway, outgoingInterface)
+				break
+			}
 			logrus.Printf("Found default gateway %v outgoing: %v", ip.String(), outgoingInterface)
 			return outgoingInterface, ip, nil
 		}
 	}
-	return "", nil, errors.New("Failed to locate default route...")
+	return "", nil, errors.New("failed to locate default route...")
 }
 
-func parseGatewayIP(defaultGateway string) net.IP {
+func parseGatewayIP(defaultGateway string) (net.IP, error) {
+	if len(defaultGateway) != 8 {
+		return nil, errors.New("failed to parse IP from string")
+	}
+	iv0, err := strconv.ParseInt(defaultGateway[0:2], 16, 32)
+	iv1, err := strconv.ParseInt(defaultGateway[2:4], 16, 32)
+	iv2, err := strconv.ParseInt(defaultGateway[4:6], 16, 32)
+	iv3, err := strconv.ParseInt(defaultGateway[6:], 16, 32)
+	if err != nil {
```

Comment:
it is a not good check because for example
Imagine the next case:
1) `iv0` has not parsed as int, `err` is not nil
2) `iv1` has  parsed as int, `err` is `nil` 

Finally, the method can return `err` as `nil`, but actually we faced a problem.

Reply:
@denis-tingajkin I'm not sure that there can be such use-case. `iv1` can be parsed as int = `0` and then `err` will be of type `ErrSyntax`, but not `nil`. `iv1` can also be the max int value when it cannot be represented by signed int, and then the error will be `ErrRange`.
It's always `int` value, but anytime parsing failed, there should be an error.
Can you share an example?

Reply:
@ivanayov please take a look https://play.golang.org/p/HwHeMvQw291

Reply:
I added error checking so that it caters for the above example. But I don't know how to solve this in a dry way other then checking for errors after each parse of the sliced string.

Reply:
@denis-tingajkin , @pringlewood The problem is that there's only one error check at the end after `iv3` and `err` is overwritten to `nill` by the last accurate value.
With a check for every value, it's working well:
https://play.golang.org/p/UsI5rrJFLA2

Reply:
@ivanayov thanks, I implemented this.

Reply:
@ivanayov exactly, that what I meant on the first comment :)